src/github.ts

```diff
@@ -79,30 +86,76 @@ interface GitHubPR {
 
 export class GitHub {
   octokit: Octokit;
+  request: Function;
+  graphql: Function;
   token: string | undefined;
   owner: string;
   repo: string;
   apiUrl: string;
   proxyKey?: string;
-  request: Function;
+  probotMode?: boolean;
 
   constructor(options: GitHubOptions) {
     this.token = options.token;
     this.owner = options.owner;
     this.repo = options.repo;
     this.apiUrl = options.apiUrl || 'https://api.github.com';
-    this.octokit = new Octokit({ baseUrl: options.apiUrl });
     this.proxyKey = options.proxyKey;
-    const defaults: { [key: string]: string | object } = {
-      baseUrl: this.apiUrl,
-      headers: {
-        'user-agent': `release-please/${require('../../package.json').version}`,
-        // some proxies do not require the token prefix.
-        Authorization: `${this.proxyKey ? '' : 'token '}${this.token}`,
-      },
-    };
-    //if (options.proxyKey) defaults['key'] = options.proxyKey;
-    this.request = request.defaults(defaults);
+
+    if (options.octokitAPIs === undefined) {
+      this.octokit = new Octokit({ baseUrl: options.apiUrl });
+      const defaults: { [key: string]: string | object } = {
+        baseUrl: this.apiUrl,
+        headers: {
+          'user-agent': `release-please/${
+            require('../../package.json').version
+          }`,
+          // some proxies do not require the token prefix.
+          Authorization: `${this.proxyKey ? '' : 'token '}${this.token}`,
+        },
+      };
+      this.request = request.defaults(defaults);
+      this.graphql = graphql;
+    } else {
+      // for the benefit of probot applications, we allow a configured instance
+      // of octokit to be passed in as a parameter.
+      this.probotMode = true;
+      this.octokit = options.octokitAPIs.octokit;
+      this.request = options.octokitAPIs.request;
+      this.graphql = options.octokitAPIs.graphql;
+    }
+  }
+
+  private async graphqlRequest(_opts: {
+    [key: string]: string | number | null | undefined;
+  }) {
+    let opts = Object.assign({}, _opts);
+    if (this.probotMode === undefined) {
+      opts = Object.assign(opts, {
```

Comment:
Can we wrap this in the default function assigned in the constructor and avoid needing probotMode?